### DISCUSSION TITLE
Installing plugins & loading automatically it in Lazy

### DISCUSSION THREAD

**Discussion by developemmanuel** [NvChad/NvChad#2008] [Show and tell]
# Issue
I was searching how to install a plugin for moving code lines ([move.nvim](https://github.com/fedepujol/move.nvim))
Once I install the plugin
```lua
-- File: custom/plugins.lua
  {
    "fedepujol/move.nvim",
  }
```
I use the Lazy option to see loaded plugins (`:Lazy`). **The plugin wasn't loaded**. So I search in the official documentation in the section about ([custom mapping](https://nvchad.com/docs/config/mappings#manually_load_mappings)) to se how load the plugin. The documentation says that we have to use
```lua
require("core.utils").load_mappings("plugin_name")
```
In order to load the mappings but it doesn't work. 
## First approach
So I loaded manually with `:Lazy load move.nvim` and it work, but the issue is that you have to load manually each time you start an `nvim` session.

# Solution
So the main solution for issue that I found is by adding this for each plugin:
```lua
-- FIle: custom/mapping.lua

-- Syntax: require("core.utils").lazy_load("plugin_name")
require("core.utils").lazy_load("move.nvim")

-- Extra example:
require("core.utils").lazy_load("Comment.nvim")
```

I hope it helped you 👍

**Comment by siduck** [comment]
@developemmanuel  or lazy=false or event="VeryLazy" 

or just look for a way to lazyload that plugin 

just put all of its mappings in its init function or make nvchad styled mappings for them

```lua
local opts = { noremap = true, silent = true }
-- Normal-mode commands
vim.keymap.set('n', '<A-j>', ':MoveLine(1)<CR>', opts)
vim.keymap.set('n', '<A-k>', ':MoveLine(-1)<CR>', opts)
vim.keymap.set('n', '<A-h>', ':MoveHChar(-1)<CR>', opts)
vim.keymap.set('n', '<A-l>', ':MoveHChar(1)<CR>', opts)
vim.keymap.set('n', '<leader>wf', ':MoveWord(1)<CR>', opts)
vim.keymap.set('n', '<leader>wb', ':MoveWord(-1)<CR>', opts)

-- Visual-mode commands
vim.keymap.set('v', '<A-j>', ':MoveBlock(1)<CR>', opts)
vim.keymap.set('v', '<A-k>', ':MoveBlock(-1)<CR>', opts)
vim.keymap.set('v', '<A-h>', ':MoveHBlock(-1)<CR>', opts)
vim.keymap.set('v', '<A-l>', ':MoveHBlock(1)<CR>', opts)
```
